### PR TITLE
Changes required for a successful build that passes tests 

### PR DESCRIPTION
**Pull request by saienduri**
This commit makes the necessary changes to get out a successful release that passes all tests.
Also added some scripts to help with release validation.
New build release CI to make sure no commits break iree-turbine release build.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def load_requirement_pins(requirements_file: str):
 
 def get_version_spec(dep: str):
     if dep in requirement_pins:
-        return f">={requirement_pins[dep]}"
+        return f"=={requirement_pins[dep]}"
```
Comment (stellaraccident) [inline setup.py:66]: Not this, please. We can't have user pins to a very specific torch version.

File: setup.py [hunk 2]
```diff
@@ -103,13 +107,14 @@ def initialize_options(self):
         f"numpy{get_version_spec('numpy')}",
         f"iree-compiler{get_version_spec('iree-compiler')}",
         f"iree-runtime{get_version_spec('iree-runtime')}",
-        "torch>=2.3.0",
+        "torch==2.3.0",
```
Comment (stellaraccident) [inline setup.py:110]: Ditto.

File: setup.py (hunk 1 above)
Comment (saienduri) [reply setup.py]: done

**Review by stellaraccident — APPROVED** [review]
Thank you.

**Comment by stellaraccident** [comment]
Can you please squash your changes then `git commit --amend -s` to sign off and pass the DCO check? Thanks.

**Pull request merged**